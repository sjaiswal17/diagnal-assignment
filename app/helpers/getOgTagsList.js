export default () => {
  return [{
    property: 'og:title',
    keyName: 'title'
  }, {
    property: 'og:type',
    keyName: 'type'
  }, {
    property: 'og:logo',
    keyName: 'logo'
  }, {
    group: true,
    property: 'og:image',
    keyName: 'image'
  }, {
    group: true,
    property: 'og:image:url',
    keyName: 'image'
  }, {
    group: true,
    property: 'og:image:secure_url',
    keyName: 'image'
  }, {
    property: 'og:url',
    keyName: 'url'
  }, {
    group: true,
    property: 'og:audio',
    keyName: 'audio'
  }, {
    group: true,
    property: 'og:audio:url',
    keyName: 'audio'
  }, {
    group: true,
    property: 'og:audio:secure_url',
    keyName: 'audio'
  }, {
    property: 'og:description',
    keyName: 'description'
  }, {
    property: 'og:determiner',
    keyName: 'determiner'
  }, {
    property: 'og:locale',
    keyName: 'locale'
  }, {
    property: 'og:locale:alternate',
    keyName: 'localeAlternate'
  }, {
    property: 'og:site_name',
    keyName: 'siteName'
  }, {
    property: 'og:product:retailer_item_id',
    keyName: 'productRetailerItemId'
  }, {
    property: 'og:product:price:amount',
    keyName: 'productPriceAmount'
  }, {
    property: 'og:product:price:currency',
    keyName: 'productPriceCurrency'
  }, {
    property: 'og:product:availability',
    keyName: 'productAvailability'
  }, {
    property: 'og:product:condition',
    keyName: 'productCondition'
  }, {
    property: 'og:price:amount',
    keyName: 'priceAmount'
  }, {
    property: 'og:price:currency',
    keyName: 'priceCurrency'
  }, {
    property: 'og:availability',
    keyName: 'availability'
  }, {
    group: true,
    property: 'og:video',
    keyName: 'video'
  }, {
    group: true,
    property: 'og:video:url', // An alternative to 'og:video'
    keyName: 'video'
  }, {
    group: true,
    property: 'og:video:secure_url',
    keyName: 'video'
  }]
}
