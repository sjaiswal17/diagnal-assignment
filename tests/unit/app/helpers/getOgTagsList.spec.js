import getOgTagsList from '../../../../app/helpers/getOgTagsList'

describe('helpers: getOgTagsList: Get the list of OG tags', () => {
  it('it should return an array', () => {
    const result = getOgTagsList()
    expect(result).toMatchSnapshot()
  })
})
