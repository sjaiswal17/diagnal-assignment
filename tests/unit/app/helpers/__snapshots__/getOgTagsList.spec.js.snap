// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers: getOgTagsList: Get the list of OG tags it should return an array 1`] = `
Array [
  Object {
    "keyName": "title",
    "property": "og:title",
  },
  Object {
    "keyName": "type",
    "property": "og:type",
  },
  Object {
    "keyName": "logo",
    "property": "og:logo",
  },
  Object {
    "group": true,
    "keyName": "image",
    "property": "og:image",
  },
  Object {
    "group": true,
    "keyName": "image",
    "property": "og:image:url",
  },
  Object {
    "group": true,
    "keyName": "image",
    "property": "og:image:secure_url",
  },
  Object {
    "keyName": "url",
    "property": "og:url",
  },
  Object {
    "group": true,
    "keyName": "audio",
    "property": "og:audio",
  },
  Object {
    "group": true,
    "keyName": "audio",
    "property": "og:audio:url",
  },
  Object {
    "group": true,
    "keyName": "audio",
    "property": "og:audio:secure_url",
  },
  Object {
    "keyName": "description",
    "property": "og:description",
  },
  Object {
    "keyName": "determiner",
    "property": "og:determiner",
  },
  Object {
    "keyName": "locale",
    "property": "og:locale",
  },
  Object {
    "keyName": "localeAlternate",
    "property": "og:locale:alternate",
  },
  Object {
    "keyName": "siteName",
    "property": "og:site_name",
  },
  Object {
    "keyName": "productRetailerItemId",
    "property": "og:product:retailer_item_id",
  },
  Object {
    "keyName": "productPriceAmount",
    "property": "og:product:price:amount",
  },
  Object {
    "keyName": "productPriceCurrency",
    "property": "og:product:price:currency",
  },
  Object {
    "keyName": "productAvailability",
    "property": "og:product:availability",
  },
  Object {
    "keyName": "productCondition",
    "property": "og:product:condition",
  },
  Object {
    "keyName": "priceAmount",
    "property": "og:price:amount",
  },
  Object {
    "keyName": "priceCurrency",
    "property": "og:price:currency",
  },
  Object {
    "keyName": "availability",
    "property": "og:availability",
  },
  Object {
    "group": true,
    "keyName": "video",
    "property": "og:video",
  },
  Object {
    "group": true,
    "keyName": "video",
    "property": "og:video:url",
  },
  Object {
    "group": true,
    "keyName": "video",
    "property": "og:video:secure_url",
  },
]
`;
